app/modules/users/permissions/rules.py

```diff
@@ -85,15 +80,18 @@ def __init__(self, module=None, action=AccessOperation.READ, **kwargs):
 
     def check(self):
         from app.modules.submissions.models import Submission
+        from app.modules.users.models import User
 
         # This Rule is for checking permissions on modules, so there must be one,
         assert self._module is not None
+        log.info('Checking module %s, action %s' % (self._module, self._action))
+        log.info('current user %s' % current_user)
```

Comment:
Note it is this line that causes the test failure because it accesses current_user but it only fails on write for the User and only fails on the options checking, not the call. The exceptional thing about user write is that it does not have an @api.login_required decorator. 
Moving the delete api.login_required decorator fixed that problem